packages/app/components/TokenIcon.tsx

```diff
@@ -0,0 +1,107 @@
+import { isAddress } from "viem";
+import { cva } from "class-variance-authority";
+import { Token } from "@/models/order";
+import Image from "next/image";
+import { useCallback, useEffect, useState } from "react";
+import { useNetwork } from "wagmi";
+import { twMerge } from "tailwind-merge";
+
+interface TokenIconProps {
+  token: Token;
+  className?: string;
+  size?: "lg" | "md" | "sm" | "xs";
+}
+
+const TOKEN_LIST_BY_CHAIN_URL: { [chainId: number]: string } = {
+  1: "https://tokens.1inch.eth.link/",
+  100: "https://tokens.honeyswap.org/",
+};
+
+type TokenFromList = {
+  address: string;
+  chainId: number;
+  decimals: number;
+  logoURI: string;
+  name: string;
+  symbol: string;
+};
+
+const GNOSIS_CHAIN_ID = 100;
+
+export const TokenIcon = ({ token, className, size }: TokenIconProps) => {
+  const { chain } = useNetwork();
+  const [tokensList, setTokensList] = useState<TokenFromList[]>([]);
+
+  const setupTokenList = useCallback(async () => {
+    async function getTokensListData() {
+      const res = await fetch(
+        chain
+          ? TOKEN_LIST_BY_CHAIN_URL[chain.id]
+          : TOKEN_LIST_BY_CHAIN_URL[GNOSIS_CHAIN_ID]
```

Comment:
Should we fetch if there is no chain?

Reply:
What do you mean by no chain? The user isn't connect to a chain?

If so, yes, I think we need to have a default chain when user is not connected.